code/modules/admin/battle_royale.dm

```diff
@@ -0,0 +1,429 @@
+		radius --
+	if(process_num > 1000 && prob(0.5))
+		generate_endgame_drop()
+
+//==================================
+// INITIALIZATION
+//==================================
+
+/datum/battle_royale_controller/proc/start()
+	//Give Verbs to admins
+	for(var/client/C in GLOB.admins)
+		if(check_rights_for(C, R_FUN))
+			C.verbs += BATTLE_ROYALE_AVERBS
+	toggle_ooc(FALSE)
+	to_chat(world, "<span class='ratvar'><font size=24>Battle Royale will begin soon...</span></span>")
+	//Stop new player joining
+	GLOB.enter_allowed = FALSE
+	world.update_status()
+	if(SSticker.current_state < GAME_STATE_PREGAME)
+		to_chat(world, "<span class=boldannounce>Battle Royale: Waiting for server to be ready...</span>")
+		SSticker.start_immediately = FALSE
+		UNTIL(SSticker.current_state >= GAME_STATE_PREGAME)
+		to_chat(world, "<span class=boldannounce>Battle Royale: Done!</span>")
+	//Delay pre-game if we are in it
+	if(SSticker.current_state == GAME_STATE_PREGAME)
+		//Force people to be not ready and start the game
+		for(var/mob/dead/new_player/player in GLOB.player_list)
+			to_chat(player, "<span class=greenannounce>You have been forced as an observer. When the prompt to join battle royale comes up, press yes. This is normal and you are still in queue to play.</span>")
+			player.ready = FALSE
+			player.make_me_an_observer(TRUE)
+		to_chat(world, "<span class='boldannounce'>Battle Royale: Force-starting game.</span>")
+		SSticker.start_immediately = TRUE
+	SEND_SOUND(world, sound('sound/misc/server-ready.ogg'))
+	sleep(50)
+	//Clear client mobs
+	to_chat(world, "<span class='boldannounce'>Battle Royale: Clearing world mobs.</span>")
+	for(var/mob/M as anything in GLOB.player_list)
+		if(isliving(M))
+			qdel(M)
+		CHECK_TICK
+	sleep(50)
+	to_chat(world, "<span class='greenannounce'>Battle Royale: STARTING IN 30 SECONDS.</span>")
+	to_chat(world, "<span class='greenannounce'><i>If you are on the main menu, observe immediately to sign up. (You will be prompted in 30 seconds.)</i></span>")
+	toggle_ooc(TRUE)
+	sleep(300)
+	toggle_ooc(FALSE)
+	to_chat(world, "<span class='boldannounce'>Battle Royale: STARTING IN 5 SECONDS.</span>")
+	to_chat(world, "<span class='greenannounce'>Make sure to hit yes to the sign up message given to all observing players.</span>")
+	sleep(50)
+	to_chat(world, "<span class='boldannounce'>Battle Royale: Starting game.</span>")
+	titanfall()
+	death_wall = list()
+	var/z_level = SSmapping.station_start
+	var/turf/center = SSmapping.get_station_center()
+	var/list/edge_turfs = list()
+	edge_turfs += block(locate(12, 12, z_level), locate(244, 12, z_level))			//BOTTOM
+	edge_turfs += block(locate(12, 244, z_level), locate(244, 244, z_level))		//TOP
+	edge_turfs |= block(locate(12, 12, z_level), locate(12, 244, z_level))			//LEFT
+	edge_turfs |= block(locate(244, 12, z_level), locate(244, 244, z_level)) 	//RIGHT
+	for(var/turf/T in edge_turfs)
+		var/obj/effect/death_wall/DW = new(T)
+		DW.set_center(center)
+		death_wall += DW
```

Comment:
Feel like you could use a few check_ticks in here.

Reply:
Added that and moved the start of processing to after it to prevent the wall from being updated to early.